workflows/data_io_utils/mgnify_v6_utils/amplicon.py

```diff
@@ -0,0 +1,199 @@
+        return
+
+    tax_dir.files.append(
+        File(
+            path=tax_dir.path
+            / f"{analysis.run.first_accession}_{fs_label_for_tax_source}.tsv",
+            rules=[
+                FileExistsRule,
+                FileIsNotEmptyRule,
+                FileConformsToTaxonomyTSVSchemaRule,
+            ],
+        )
+    )
+
+    with tax_dir.files[0].path.open("r") as tax_tsv:
+        move_file_pointer_past_comment_lines(
+            tax_tsv, delimiter=CSVDelimiter.TAB, comment_char="#"
+        )
+        try:
+            tax_df = pd.read_csv(tax_tsv, sep=CSVDelimiter.TAB)
+        except pd.errors.EmptyDataError:
+            logging.error(
+                f"Found empty taxonomy TSV in {tax_dir}. Probably unit testing, otherwise we should never be here"
+            )
+            return
+
+    tax_df = tax_df.rename(columns={"taxonomy": "organism"})
+    tax_df["count"] = None
+    tax_df: pd.DataFrame = tax_df[["organism", "count"]]
+
+    taxonomies = analysis.annotations.get(analysis.TAXONOMIES, {})
+    if not taxonomies:
+        taxonomies = {}
+    taxonomies[analysis.TaxonomySources.SSU.value] = tax_df.to_dict(orient="records")
+    analysis.annotations[analysis.TAXONOMIES] = taxonomies
+
+    tax_dir.files.append(
+        File(
+            path=tax_dir.path / f"{analysis.run.first_accession}.html",
+            rules=[
+                FileExistsRule,
+                FileIsNotEmptyRule,
+            ],
+        )
+    )
+    analysis.add_download(
+        DownloadFile(
+            path=tax_dir.files[-1].path.relative_to(analysis.results_dir),
+            file_type=DownloadFileType.HTML,
+            alias=f"{analysis.run.first_accession}_{source.value}.html",
+            download_type=DownloadType.TAXONOMIC_ANALYSIS,
+            parent_identifier=analysis.accession,
+            short_description=f"{source.value} Krona plot",
+            long_description=f"Krona plot webpage showing taxonomic assignments from {source.value} annotation",
+        )
+    )
+    analysis.save()
+
+
+def import_qc(
+    analysis: analyses.models.Analysis,
+    dir_for_analysis: Path,
+    allow_non_exist: bool = True,
+):
+    if not allow_non_exist:
+        qc_dir = Directory(
+            path=dir_for_analysis  # /hps/prod/...../abc123/SRR999/
+            / EMG_CONFIG.amplicon_pipeline.qc_folder,  # qc/
+            rules=[DirectoryExistsRule],
+            glob_rules=[GlobOfQcFolderHasFastpAndMultiqc],
+        )
+    else:
+        qc_dir = Directory(
+            path=dir_for_analysis  # /hps/prod/...../abc123/SRR999/
+            / EMG_CONFIG.amplicon_pipeline.qc_folder  # qc/
+        )
+
+    if not qc_dir.path.is_dir():
+        print(f"No qc dir at {qc_dir.path}. Nothing to import.")
+        return
+
+    # TODO rules and downloadfile for multiqc
+    qc_dir.files.append(
+        File(
+            path=qc_dir.path / f"{analysis.run.first_accession}_multiqc_report.tsv",
```

Comment:
Should those filenames be also in config or somewhere else in one place defined? 
I think for tools like fastp or multiqc that we will use in many/all our pipelines we can re-use names of files (that will help also developing other pipelines with the same filenames/structures).